master/buildbot/reporters/mail.py

```diff
@@ -269,9 +273,15 @@ def matchesAnyTag(self, tags):
     def isMailNeeded(self, build):
         # here is where we actually do something.
         builder = build['builder']
+        scheduler = build['properties']['scheduler'][0]
```

Comment:
I dont think that we are garanteed to have scheduler and branch properties.
please use 

```
build['properties'].get('scheduler',[None])[0]
```
and same for branch